src/System.Private.CoreLib/shared/System/Enum.cs

```diff
@@ -322,28 +323,29 @@ internal static ulong ToUInt64(object value)
 
         public static object Parse(Type enumType, string value, bool ignoreCase)
         {
-            bool success = TryParse(enumType, value, ignoreCase, throwOnFailure: true, out object result);
+            bool success = TryParse(enumType, value, ignoreCase, throwOnFailure: true, out object? result);
             Debug.Assert(success);
-            return result;
+            // TODO-NULLABLE: workaround compiler bug
```

Comment:
What's the compiler bug?  The compiler can't know that TryParse is always going to either return true or throw, and even if it did, without annotations on TryParse that say that `result` will be non-null if the method returns true, there's no way the compiler could know that `result` won't be null here.  Seems like this `!` is here for good.

Reply:
Well when throwOnFailure: true result will not be null, that was the point, but can't really say it is doable by compiler though. Will remove the comment